src/Microsoft.AspNetCore.SignalR.Client.Core/IHubConnectionBuilder.cs

```diff
@@ -3,14 +3,17 @@
 
 using System;
 using Microsoft.AspNetCore.Connections;
+using Microsoft.Extensions.DependencyInjection;
 
 namespace Microsoft.AspNetCore.SignalR.Client
 {
     public interface IHubConnectionBuilder
```

Comment:
This should implement ISignalRBuilder (ISignalRBuilder should be moved into common)

Reply:
The problem with that is all of the `ISignalRBuilder` extension methods don't apply to the client. The only ones which kind of make sense are `AddJsonProtocol` and `AddMessagePackProtocol`, but they build an enumerable of protocols, while the client only makes sense with one.

Intellisense will be filled with extension methods that will confuse any devs that brought in non-client packages into their project.

Reply:
Then I think we should consider doing ISignaRClientBuilder : ISignalRBuilder. The intent being that something will be shared (like the protocol) and should not require a developer to write 2 packages.

Reply:
Yep

Reply:
Ok, but what is the benefit of sharing the `ISignalRBuilder` interface is nothing uses it directly.

Reply:
Yeah, it's a bit unclear to me what we would share. The current implementation uses `TryAddEnumerable` which means you could get in weird order-dependent situations if you try to add both.

Reply:
The protocols will use them directly and we will remove the 2 msg pack dlls and just have a single one.